rfcs/table-destructuring.md

```diff
@@ -0,0 +1,236 @@
+# Table Destructuring
+
+## Summary
+A feature present and highly-used in JavaScript, table destructuring allows the developer to extract data from tables and assign them to variables in a clean, shortened manner, thus avoiding the need for excessive index calls and several individual assignments, most of which being expendable.
```

Comment:
JavaScript's destructuring is mentioned here as being highly used, but the destructuring described later in the RFC has barely any similarities to JavaScript. For example:
```js
let {a="b"} = {}
```
In JavaScript, equals followed by a value will provided a default value if nothing is found in the object (which has no equivalent in this RFC). In this RFC, it is used for using an alternative name and destructuring a value within the table further. Why diverge in such a significant way from the language which inspired this RFC?